File: docs/DSA/src/py/valid_parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stk = []
        op = {"(": ")", "{": "}", "[": "]"}
        for ch in s:
            if ch in op:
                stk.append(ch)
            else:
                if len(stk) == 0 or (op[stk[-1]] != ch):
                    return False
                stk.pop()
        return len(stk) == 0

File: docs/DSA/src/go/valid_parentheses.go
package main

func isValid(s string) bool {
	var stk []rune
	op := map[rune]rune{'(': ')', '{': '}', '[': ']'}
	for _, ch := range s {
		val, ok := op[ch]
		if ok {
			stk = append(stk, val)
		} else {
			if len(stk) == 0 || stk[len(stk)-1] != ch {
				return false
			}
			stk = stk[:len(stk)-1]
		}
	}
	return len(stk) == 0
}
